include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -3566,17 +3566,10 @@ CellAccessor<dim, spacedim>::is_locally_owned() const
 #ifndef DEAL_II_WITH_MPI
   return true;
 #else
-  if (is_artificial())
-    return false;
-
-  const parallel::TriangulationBase<dim, spacedim> *pt =
-    dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
-      this->tria);
 
-  if (pt == nullptr)
-    return true;
-  else
-    return (this->subdomain_id() == pt->locally_owned_subdomain());
+  return (this->tria->locally_owned_subdomain() ==
+            numbers::invalid_subdomain_id ||
```

Comment:
Can I ask you to prefix all of these comparisons with a comment that says that serial triangulations report `numbers::invalid_subdomain_id` for `tria->locally_owned_subdomain()`, and that consequently the first part of the condition indicates that the triangulation is serial where all cells are locally owned?

Same in the other places.